commit 6fa90611b7402ce0231368af67528fdab6419250
Author: Rafael Garcia-Suarez <rgarciasuarez@mandriva.org>
Date:   Thu Mar 16 14:19:14 2006 +0000

    Don't search perl files for provides if they don't end with .pm (because
    if they don't, the perl interpreter won't look them up in @INC)

diff --git a/scripts/perl.prov b/scripts/perl.prov
index 6b6774a..68ff2c8 100755
--- a/scripts/perl.prov
+++ b/scripts/perl.prov
@@ -47,6 +47,7 @@
 
 if ("@ARGV") {
   foreach (@ARGV) {
+    next if !/\.pm$/;
     process_file($_);
   }
 } else {
@@ -55,6 +56,7 @@ if ("@ARGV") {
   # contents of the file.
 
   foreach (<>) {
+    next if !/\.pm$/;
     process_file($_);
   }
 }
